src/pod.ts

```diff
@@ -1471,3 +1517,233 @@ export class PodScheduling {
+
+  /**
+   * Ports to allow incoming traffic to.
+   *
+   * @default - The pod ports.
+   */
+  readonly ports?: networkpolicy.NetworkPolicyPort[];
+
+}
+
+/**
+ * Controls network isolation rules for inter-pod communication.
+ */
+export class PodConnections {
+
+  constructor(protected readonly instance: AbstractPod) {}
+
+  /**
+   * Allow network traffic from this pod to the peer.
+   *
+   * By default, this will create an egress network policy for this pod, and an ingress
+   * network policy for the peer. This is required if both sides are already isolated.
+   * Use `options.isolation` to control this behavior.
+   *
+   * @example
+   *
+   * // create only an egress policy that selects the 'web' pod to allow outgoing traffic
+   * // to the 'redis' pod. this requires the 'redis' pod to not be isolated for ingress.
+   * web.connections.allowTo(redis, { isolation: Isolation.POD })
+   *
+   * // create only an ingress policy that selects the 'redis' peer to allow incoming traffic
+   * // from the 'web' pod. this requires the 'web' pod to not be isolated for egress.
+   * web.connections.allowTo(redis, { isolation: Isolation.PEER })
+   *
+   */
+  public allowTo(peer: networkpolicy.INetworkPolicyPeer, options: PodConnectionsAllowToOptions = {}) {
+    return this.allow('Egress', peer, { ports: this.extractPorts(peer), ...options });
+  }
+
+  /**
+   * Allow network traffic from the peer to this pod.
+   *
+   * By default, this will create an ingress network policy for this pod, and an egress
+   * network policy for the peer. This is required if both sides are already isolated.
+   * Use `options.isolation` to control this behavior.
+   *
+   * @example
+   *
+   * // create only an egress policy that selects the 'web' pod to allow outgoing traffic
+   * // to the 'redis' pod. this requires the 'redis' pod to not be isolated for ingress.
+   * redis.connections.allowFrom(web, { isolation: Isolation.PEER })
+   *
+   * // create only an ingress policy that selects the 'redis' peer to allow incoming traffic
+   * // from the 'web' pod. this requires the 'web' pod to not be isolated for egress.
+   * redis.connections.allowFrom(web, { isolation: Isolation.POD })
+   *
+   */
+  public allowFrom(peer: networkpolicy.INetworkPolicyPeer, options: PodConnectionsAllowFromOptions = {}) {
+    return this.allow('Ingress', peer, { ports: this.extractPorts(this.instance), ...options });
+  }
+
+  private allow(direction: 'Ingress' | 'Egress', peer: networkpolicy.INetworkPolicyPeer, options: PodConnectionsAllowToOptions | PodConnectionsAllowFromOptions = {}) {
+
+    const config = peer.toNetworkPolicyPeerConfig();
+    networkpolicy.validatePeerConfig(config);
+
+    const peerAddress = this.peerAddress(config);
+
+    if (!options.isolation || options.isolation === PodConnectionsIsolation.POD) {
+
+      const src = new networkpolicy.NetworkPolicy(this.instance, `Allow${direction}${peerAddress}`, {
```

Comment:
Adde a unit test for different inputs. As far as the same input, it will currently throw a "Construct already exists" error, which I know isn't great, but we can address this later if needed.